### DISCUSSION TITLE
Transaction -> Stakeholder linking

### DISCUSSION THREAD

**Discussion by haoting-addepar** [Open-Cap-Table-Coalition/Open-Cap-Format-OCF#208] [Q&A] [answered]
Hi, I would like some clarification on linking a transaction to its stakeholder. Looking at the spec, the only transaction types that have `stakeholder_id` are issuance. All other transaction types only have a `security_id` available for linking.

Here are a couple of questions I have

1. Will an issuance transaction always generate a unique security id even for the same stakeholder. For example, I work for private company X, and I will have new equity refreshes every year. Will this use case map into multiple stock issuance tx, and each one will have the same `stakeholder_id` (pointing to me) but a unique `security_id`.

2. How would I know who the counter-party is for a transfer tx? The spec has a `resulting_security_ids` field, but how do I tell who the stakeholders are for these resulting security ids?

Thanks

**Comment by JSv4** [comment, marked as answer]
Hey @haoting-addepar, thanks for the questions. Quick thoughts below:

1. Yes, each issuance should generate a new, unique security_id. In your example, it's sort of analogous to how, with certificated shares, you'd get a new certificate (with its own certificate number) for each new tranche of stock issued. You can obviously aggregate the issuances if you want to only see your aggregate holdings, but the underlying OCF will represent the precise history of what was issued when, with each issuance having a unique ID.
2. As above, we thought through this by analogizing to the way things were done in the paper days. Typically, you'd issue a new certificate to the transferee (let's call it certificate B) and then cancel the certificate held by the transferor (let's call that certificate A) and mark somewhere that the shares represented by certificate A transferred to certificate B. To  model a transfer in OCF, you should create a new issuance event with the new stakeholder information. Then create an accompanying transfer event that links the old issuance to the new one. If there's no `balance_security_id` on the transfer event, that means no shares from the original issuance remain held by the original stakeholder. We actually just had a [discussion on how to model this](https://github.com/Open-Cap-Table-Coalition/Open-Cap-Format-OCF/discussions/192) and have a pending issue to improve the documentation.

**Reply by haoting-addepar** [reply]
Great. thank you for the quick response.

**Reply by JSv4** [reply]
Anytime!
